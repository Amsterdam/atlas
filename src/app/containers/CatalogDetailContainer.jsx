import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { AngularWrapper } from 'react-angular';
import { isDetailLoading } from '../../shared/ducks/detail/detail';
import { getUser } from '../../shared/ducks/user/user';
import { getCatalogFilters } from '../../catalog/ducks/data-selection/data-selection-catalog';

const mapStateToProps = (state) => ({
  isLoading: isDetailLoading(state),
  catalogFilters: getCatalogFilters(state),
  user: getUser(state),
  endpoint: `https://acc.api.data.amsterdam.nl/dcatd/datasets/${state.catalog.detail}` // TODO: refactor use API_ROOT and such
});

const CatalogDetailContainer = ({
  isLoading,
  catalogFilters,
  user,
  endpoint
}) => (
  <div className="qa-detail">
    <AngularWrapper
      moduleName={'dpDetailWrapper'}
      component="dpDetail"
      dependencies={['atlas']}
      bindings={{
        isLoading,
        catalogFilters,
        user
      }}
      interpolateBindings={{
        endpoint
      }}
    />
  </div>
);

CatalogDetailContainer.defaultProps = {
  isLoading: false
};

CatalogDetailContainer.propTypes = {
  isLoading: PropTypes.bool,
  catalogFilters: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  user: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  endpoint: PropTypes.string.isRequired
};

export default connect(mapStateToProps, null)(CatalogDetailContainer);
