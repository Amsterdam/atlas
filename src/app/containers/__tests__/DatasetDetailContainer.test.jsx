import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import CatalogDetailContainer from '../DatasetsDetailContainer';

const initialState = {
  detail: {
    isLoading: false
  },
  catalog: {
    detail: 'foo'
  },
  datasets: {
    datasetApiSpecification: {
      data: {}
    }
  },
  user: {},
  location: {
    payload: {
      id: 1
    }
  }
};

describe('CatalogDetailContainer', () => {
  it('should render', () => {
    const store = configureMockStore()({ ...initialState });
    const component = shallow(<CatalogDetailContainer />, { context: { store } }).dive();
    expect(component).toMatchSnapshot();
  });
});
