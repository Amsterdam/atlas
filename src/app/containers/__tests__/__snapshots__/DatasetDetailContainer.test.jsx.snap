// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetsDetailContainer should render 1`] = `
<div
  className="c-dashboard__content qa-detail"
>
  <AngularWrapper
    bindings={
      Object {
        "catalogFilters": Object {},
        "isLoading": false,
        "user": Object {},
      }
    }
    component="dpDetail"
    dependencies={
      Array [
        "atlas",
      ]
    }
    interpolateBindings={
      Object {
        "endpoint": "https://acc.api.data.amsterdam.nl/dcatd/datasets/1",
      }
    }
    moduleName="dpDetailWrapper"
  />
</div>
`;
